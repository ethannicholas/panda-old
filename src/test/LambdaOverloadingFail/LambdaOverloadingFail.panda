method doSomething(fn:(Int)=>(Int)) {
    Console.writeLine("I am doing something to 42: \{fn(42)}")
}

method doSomething(fn:(Real)=>(Real)) {
    Console.writeLine("I am doing something real to 42: \{fn(42)}")
}

doSomething(x => x * 2)