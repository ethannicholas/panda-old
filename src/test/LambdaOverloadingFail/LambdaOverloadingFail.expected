error: LambdaOverloadingFail.panda:9:1: call to 'doSomething' with parameters (<lambda>) is ambiguous. The following methods are all equally good matches:
    method doSomething(fn:(panda.core.Int32)=>(panda.core.Int32)) (LambdaOverloadingFail.panda:1:1)
    method doSomething(fn:(panda.core.Real64)=>(panda.core.Real64)) (LambdaOverloadingFail.panda:5:1)
1 error
